archaius2-core/src/main/java/com/netflix/archaius/ConfigProxyFactory.java

```diff
@@ -206,21 +215,29 @@ public T invoke(Object Obj, Object[] args) {
                     verb = "";
                 }
                 
+                final Class<?> returnType = m.getReturnType();
+                
                 Object defaultValue = null;
                 if (m.getAnnotation(DefaultValue.class) != null) {
                     if (m.isDefault()) {
                         throw new IllegalArgumentException("@DefaultValue cannot be defined on a method with a default implementation for method "
                                 + m.getDeclaringClass().getName() + "#" + m.getName());
+                    } else if (
+                            Map.class.isAssignableFrom(returnType) ||
+                            List.class.isAssignableFrom(returnType) ||
+                            Set.class.isAssignableFrom(returnType) ) {
+                        throw new IllegalArgumentException("@DefaultValue cannot be used with collections.  Use default method implemenation instead "
```

Comment:
To properly support this I'd need to make more changes than I'm comfortable making in this one PR.  I'd rather just support defaults via default methods and add support for @DefaultValue at a later time.